### ISSUE TITLE
biplots is not working due to input validation

### ISSUE THREAD

**Issue by antgonza** [biocore/emperor#384]
This [line](https://github.com/biocore/emperor/blame/master/scripts/make_emperor.py#L545) checks that the sample names and the summarized taxonomies overlap, which should not be the case. The taxonomic summaries remove the sample names.

cc @wasade @ElDeveloper

**Comment by ElDeveloper** [comment]
If you are collapsing several samples into a single category, then this
is an expected behavior as we don't have a way to match any of the data
to the coordinates matrix. Is this not the use-case you are describing?

**Comment by antgonza** [comment]
My first report was based on what I was _assuming_ it was happening but after further review it seems that the problem is that the OTU table has more samples than the pc/map files. For example, for AGP we do taxa summaries on the non rarefied table (all samples) but PCoA on rarefied one (missing some), thus failing on that test.

**Comment by ElDeveloper** [comment]
Would you advocate for filtering out the samples from the table and
raising a warning? Or just error out with a list of sample ids that you
have to filter?

**Comment by antgonza** [comment]
I thought the table was being filtered before calculating the biplot,
if not, that's a bug ...

**Comment by ElDeveloper** [comment]
Since we have this restriction in place and we sort the table using the
coordinates matrix, we don't have to filter.

https://github.com/biocore/emperor/blob/master/emperor/biplots.py#L126

**Comment by antgonza** [comment]
Got it but note that the error message is completely misleading and
doesn't tell you how to solve it:
"The sample identifiers in the OTU table must have at least one match
with the data in the mapping file and with the coordinates file.
Verify you are using input files that belong to the same dataset."

IMOO the best will be to filter ...

**Comment by ElDeveloper** [comment]
It makes sense, what do you think about a warning? Can you rephrase the
message in a way it makes more sense? Or better yet, do you want to put
this in place? ;)

**Comment by antgonza** [comment]
Sorry too busy. Agree on better message but I still believe filtering
is the real solution ...

**Comment by ElDeveloper** [comment]
@antgonza I have been thinking about this, and the problem is that if you filter the table, you would also have to re-calculate the relative abundances of the taxa for the given samples, otherwise the information would be spurious. So I think there's not really anything we can do and the error message should suggest that you filter the table and you then summarize. Do you agree or perhaps I'm missing something.

**Comment by antgonza** [comment]
Good point and agree!

**Comment by ElDeveloper** [comment]
Great, then I've relabeled this issue as a "quick fix", in case there's
any takers.

**Comment by ElDeveloper** [comment]
This is no longer an issue, since the calculation of biplots has been deferred to be in qiime2 or skbio.
